apps/testingaccessibility/src/components/team/buy-more-seats.tsx

```diff
@@ -1,21 +1,84 @@
 import React from 'react'
+import {z} from 'zod'
+import type {FormattedPrice} from '@skillrecordings/commerce-server/dist/@types'
+import {useQuery} from 'react-query'
+import {useDebounce} from '@skillrecordings/react'
 
-const BuyMoreSeats = () => {
-  const handleOnSubmit = () => {}
-  const calculatedPrice = 0
+const buildFormActionPath = (params: {
+  userId: string
+  quantity: number
+  productId: string
+}) => {
+  const {productId, quantity, userId} = params
+
+  const queryParamString = new URLSearchParams({
+    productId,
+    quantity: String(quantity),
+    userId,
+  }).toString()
+
+  return `/api/skill/checkout/stripe?${queryParamString}`
+}
+
+const buyMoreSeatsSchema = z.object({productId: z.string(), userId: z.string()})
+type BuyMoreSeatsProps = z.infer<typeof buyMoreSeatsSchema>
+
+const BuyMoreSeats = (props: BuyMoreSeatsProps) => {
+  const {productId, userId} = buyMoreSeatsSchema.parse(props)
+
+  const [quantity, setQuantity] = React.useState(5)
+  const debouncedQuantity: number = useDebounce<number>(quantity, 250)
+
+  const {data: formattedPrice, status} = useQuery<FormattedPrice>(
+    ['pricing', debouncedQuantity, productId],
+    () =>
+      fetch('/api/skill/prices', {
+        method: 'POST',
+        body: JSON.stringify({
+          productId,
+          quantity: debouncedQuantity,
+          purchases: [],
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      })
+        .then((res) => res.json())
+        .then((formattedPrice: FormattedPrice) => {
+          return formattedPrice
+        }),
+  )
+
+  const {calculatedPrice} = z
+    .object({calculatedPrice: z.number(), id: z.string()})
+    .or(z.undefined())
+    .transform((value) => {
+      return !!value ? value : {calculatedPrice: 0}
+    })
+    .parse(formattedPrice)
+
+  const formActionPath = buildFormActionPath({
+    userId,
+    quantity: debouncedQuantity,
+    productId,
+  })
 
   return (
-    <form className="pt-3" onSubmit={handleOnSubmit}>
+    <form className="pt-3" action={formActionPath} method="POST">
```

Comment:
The 'Buy' button POSTs the form to our stripe checkout API which facilitates the transition to the Stripe Checkout Session.